### PR TITLE
Otel improvements

### PR DESCRIPTION
**Pull request by markphelps**
Fixes: FLI-194

Adds some telemetry span attributes for the `Evaluate` and `GetFlag` endpoints

CleanShot 2023-01-30 at 13 16 15

CleanShot 2023-01-30 at 13 17 02

### CODE REVIEW

File: internal/server/otel/attributes.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+var (
+	AttributeMatch       = attribute.Key("flipt.match")
+	AttributeFlag        = attribute.Key("flipt.flag")
+	AttributeFlagEnabled = attribute.Key("flipt.flag_enabled")
+	AttributeSegment     = attribute.Key("flipt.segment")
+	AttributeReason      = attribute.Key("flipt.reason")
+	AttributeValue       = attribute.Key("flipt.value")
+	AttributeEntityID    = attribute.Key("flipt.entity_id")
+	AttributeRequestID   = attribute.Key("flipt.request_id")
+	AttributeDurationMS  = attribute.Key("flipt.duration_ms")
```
Comment (GeorgeMac) [inline internal/server/otel/attributes.go:14]: Food for thought: going forward I wonder if we can reconcile these: https://github.com/flipt-io/flipt/blob/main/internal/server/metrics/metrics.go#L54-L60

For now, I dont think it is worth it. Just a shame as there is some overlap.
I wish I had namespaced those `metrics.go` ones like you have.

**Review by GeorgeMac — APPROVED** [review]
Sweet 💪

File: internal/server/evaluator.go [hunk 2]
```diff
@@ -23,6 +26,27 @@ func (s *Server) Evaluate(ctx context.Context, r *flipt.EvaluationRequest) (*fli
...
+		fliptotel.AttributeRequestID.String(r.RequestId),
+	}
+
+	if resp != nil {
+		spanAttrs = append(spanAttrs,
+			fliptotel.AttributeMatch.Bool(resp.Match),
+			fliptotel.AttributeSegment.String(resp.SegmentKey),
+			fliptotel.AttributeValue.String(resp.Value),
+			fliptotel.AttributeReason.String(resp.Reason.String()),
+			fliptotel.AttributeDurationMS.Float64(resp.RequestDurationMillis),
```
Comment (GeorgeMac) [inline internal/server/evaluator.go:42]: Quick one: while I have you. Maybe this one should be dropped:
```suggestion
```

Duration is something which should be captured naturally through the span start and end timestamps. Also, it is something we dont really want to index like other attributes.

**Review by GeorgeMac — CHANGES_REQUESTED** [review]
One last minute consideration.

**Review by GeorgeMac — APPROVED** [review]

File: internal/server/otel/attributes.go (hunk 1 above)
Comment (markphelps) [reply internal/server/otel/attributes.go]: yeah i thought about that, the only difference right now is the namespace I think

**Pull request merged**